src/array/dictionary/mod.rs

```diff
@@ -127,8 +128,12 @@ impl<K: DictionaryKey> DictionaryArray<K> {
     /// Returns the value of the [`DictionaryArray`] at position `i`.
     #[inline]
     pub fn value(&self, index: usize) -> Box<dyn Scalar> {
-        let index = self.keys.value(index).to_usize().unwrap();
-        new_scalar(self.values.as_ref(), index)
+        if self.keys.is_null(index) {
+            Box::new(NullScalar::new())
```

Comment:
`NullScalar.data_type` yields `DataType::Null`, but the value yields a scalar of `values.data_type()`. This needs to be a Scalar of the same type as inner values (yeah, this API is still not very useful: technically the scalar should be "DictionaryScalar" for its data_type to be equal to the array's DataType :/

Reply:
Is there a fix for this atm? Now the test fail, because it tries to read from a negative (masked out) index.

Reply:
Meh, let's keep it for now like this; better than nothing. xD